### ISSUE TITLE
Dates and times in the wrong format

### ISSUE THREAD

**Issue by sabrinaweb** [johannesjo/super-productivity#3676]
### Your Environment

- Version used: App for android, version 10.1.1 downloaded from Aurora store
- Operating System and version: Android 13
- Desktop Environment: 
- Browser Name and version: 

### Expected Behavior

I should see all the dates in the same format, but some dates are formatted as dd/mm and some other dates are formatted as mm/dd. The same happens for times, I see some of them in 24-hour format and some in 12-hour format with am/pm.
Since I'm European, I expect all the dates as dd/mm and all the times in 24-hour format, otherwise it's misleading.

### Current Behavior

If I plan a task for a future date and then, from the project list, I choose a project with a planned task, I see that task with the date formatted as mm/dd. Tasks planned for today at a future time show the time in 24-hour format as expected. If I tap on the single task, however, I see the time in 12-hour format, without even saying if it's am or pm (but maybe it's me, because my screen is too small - 3.03" - and 'pm' just falls out of the viewport). Furthermore, you can see at the bottom 'created on mm/dd/yy, h:mm am (or pm).
If you tap on 'repeat/schedule' from the menu, you see tasks with dates formatted as dd/mm, but times are in 12-hour format (with am/pm in this case).

### Steps to Reproduce (for bugs)

1. My phone locale is set to British English
2. Create a new project
3. Add a new task to that project
4. Schedule that task for a future date and time
5. From the menu on the left, tap on the project (first screenshot)
6. Tap on the task (second screenshot)
7. From the menu on the left, tap on 'Repeat/Schedule' (third screenshot)

### Can you reproduce this reliably?

I just reproduced it today, to take some screenshots
![Screenshot_1](https://github.com/user-attachments/assets/274a3252-9659-4f97-915d-eeab530551e9)
![Screenshot_2](https://github.com/user-attachments/assets/c8fee58b-7b25-4371-a7d4-ae726280ded6)
![Screenshot_3](https://github.com/user-attachments/assets/2e379bbb-860f-489a-9348-a511d17ecaf8)

**Comment by johannesjo** [comment]
While I generally agree, we have limited space available. But there likely is a better way to tackle this as it is now :)

**Comment by GavoGavo** [comment]
Confirm this issue exists for me on windows desktop. 

I can enter the date using the magic helper in my local format (11 December 2024 = @11/12/2024). The app adds the task to the correct day, but the right-tab popout states it was created (if created on 1 December 2024) "12/01/2024".

This is an important bug in my opinion

**Comment by johannesjo** [comment]
@GavoGavo that sounds like a separate issue to me, if I understand you correctly. The original issue is about formatting. Yours sounds like a straight up bug to me. Could you maybe open up a separate issue for this?

**Comment by GavoGavo** [comment]
@johannesjo I can, but I didn't think it was a bug. Functionality is all totally fine, but formatting appears sometimes day/month, elsewhere month/day. 

If you say this is a bug though, I am happy to raise an issue?

**Comment by Cyber-Syntax** [comment]
I have a similar issue, but mine is an hour-long problem. When I look at the date from the console (`Sun Dec 15 2024 12:03:49 GMT+0200 (GMT+02:00)`), but when I start the app with `TZ=Europe/Istanbul`, it works as expected (`Sun Dec 15 2024 13:03:49 GMT+0300 (GMT+03:00)`). It wasn't a problem before version 10.

Mine timedatectl setup looks like this:
```bash
timedatectl                                                      ─╯
               Local time: Paz 2024-12-15 13:03:49 +03
           Universal time: Paz 2024-12-15 10:03:49 UTC
                 RTC time: Paz 2024-12-15 10:03:49
                Time zone: Europe/Istanbul (+03, +0300)
System clock synchronized: yes
              NTP service: active
          RTC in local TZ: no
```

**Comment by johannesjo** [comment]
Let's try to collect this systematically:

* Task items little clock does not show AM, PM, but is only ever in the 24h format
* the short syntax "@" helper parses a different format than the display values (@GavoGavo ?)
* unless I start the app with TZ=myTimezone all dates in the app use the wrong timezone  (@Cyber-Syntax ?)

**Comment by Cyber-Syntax** [comment]
@johannesjo Yes, but I have no idea how it use a different time zone since there is no `GMT+02:00` listed under  timedatectl as you can see.However, for more context, I'm certain that my other Chromium and electron base apps use the correct time zone without any issues. I'm using AppImage version by the way. 
I don't want to mislead you - it could be a Linux problem that I'll investigate further when I'm available.

**Comment by johannesjo** [comment]
That would be great @Cyber-Syntax ! Thank you!

**Comment by Cyber-Syntax** [comment]
@johannesjo Thanks for your understanding. I found that the my issue was with AppImage-run on NixOS.

**Comment by johannesjo** [comment]
@Cyber-Syntax do you think there is anything we can do on our end to improve the situation?

**Comment by sabrinaweb** [comment]
> Task items little clock does not show AM, PM but only the 24h format (the original issue @sabrinaweb )

Not quite right. Actually, sometimes I see the 24h format, sometimes I see AM/PM and sometimes I see the hour in 12h format but without the AM/PM. I want to see all times in 24h format and this is important to me, I'm a European shift worker and I find AM/PM misleading

**Comment by johannesjo** [comment]
I think this should be fixed now?

**Comment by aa956** [comment]
Not used this application apart from very short trial (in part because of the US datetimes).

Windows, MS Store version 12.0.2.0

Windows regional settings: English UI language, everything else Estonian (ShortTimePattern HH:mm, ShortDatePattern dd.MM.yyyy, ).

Application settings language: English

Settings/Schedule shows 24 hour times as in system locale:

![Image](https://github.com/user-attachments/assets/e1239790-550b-415a-9732-fa33d054d247)

Everything else seems to use language settings (US English).

Schedule view time shows AM/PM on today:

![Image](https://github.com/user-attachments/assets/9d5847d3-4ed5-41eb-9157-bac855c20ae2)

Create task in the schedule view uses US date format:

![Image](https://github.com/user-attachments/assets/f3382862-88cc-4974-87af-c43e586d8aa1)

Planner and boards show US format dates.

Is it possible to add English (ISO 8601 date 24h clock) translation, or for example English (UK) or English (Ireland) - these at least use dd.mm.yyyy dates and 24-hour clocks?

**Comment by emes81** [comment]
I'd also be a strong supporter of this - perhaps language options for English (US) and English (International), with the latter having the date/time formats that are in use in Anglophone countries other than the US, but also common (if not universal) for L2 users of English. 
Either that, or an option to set date/time formats.
Having at least one of these options is vital to usability, imo. No matter how hard I squint, "7/16" is never going to be recognisable as a date!

**Comment by johannesjo** [comment]
With the upcoming release users will be able to configure their locale. I hope that I will also have addressed all inconsistencies. Please open up a new issue if this isn't the case. Thank you!

**Comment by sabrinaweb** [comment]
Sorry, it isn't fixed at all.

I have my phone set to British English, I've chosen 24-hour format from a dropdown menu in the settings, I still see this:

<img width="480" height="854" alt="Image" src="https://github.com/user-attachments/assets/c08d6c2e-146b-4956-b4be-56e85b0f2a13" />

**Comment by johannesjo** [comment]
@sabrinaweb as mentioned could you please open up a separate issue for this?
